samcli/local/apigw/local_apigw_service.py

```diff
@@ -397,6 +403,78 @@ def _parse_lambda_output(lambda_output, binary_types, flask_request):
 
         return status_code, headers, body
 
+    @staticmethod
+    def _parse_v2_payload_format_lambda_output(lambda_output, binary_types, flask_request):
+        """
+        Parses the output from the Lambda Container
+
+        :param str lambda_output: Output from Lambda Invoke
+        :return: Tuple(int, dict, str, bool)
+        """
+        # pylint: disable-msg=too-many-statements
+        try:
+            json_output = json.loads(lambda_output)
+        except ValueError as ex:
+            raise LambdaResponseParseException("Lambda response must be valid json") from ex
+
+        # lambda can return any valid json response in payload format version 2.0.
+        # response can be a simple type like string, or integer
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.response
+        if isinstance(json_output, dict):
+            body = json_output.get("body") if "statusCode" in json_output else json.dumps(json_output)
+        else:
+            body = json_output
+            json_output = {}
+
+        if not body:
+            LOG.warning("Lambda returned empty body!")
+
+        status_code = json_output.get("statusCode") or 200
+        headers = LocalApigwService._merge_response_headers(
+            json_output.get("headers") or {}, json_output.get("multiValueHeaders") or {}
+        )
+
+        is_base_64_encoded = json_output.get("isBase64Encoded") or False
+
+        try:
+            status_code = int(status_code)
+            if status_code <= 0:
```

Comment:
This is definitely an edge case, but I think hypothetically a user could provide a negative status code since it's set by the Lambda function logic. It makes no sense, but it's possible to do.

Reply:
I see. In general I prefer to [EAFP](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/) than to explicitly handle the nonsensical cases.